src/models/app/app.ts

```diff
@@ -0,0 +1,96 @@
+import Page from '../templates/page';
+import MainPage from '../../pages/main/main';
+import DescriptionPage from '../../pages/description/description';
+import CartPage from '../../pages/cart/cart';
+import Header from '../components/header';
+import ErrorPage, { ErrorTypes } from '../../pages/404/404';
+import Footer from '../components/footer';
+import { Filters } from '../enums/filters';
+import { listener } from '../../pages/confirm/confirm';
+
+export const enum PageIds {
+  MainPage = 'main-page',
+  DescriptionPage = 'description-page',
+  ConfirmPage = 'confirm-page',
+  CartPage = 'cart-page',
+}
+
+class App {
+  private static root: HTMLElement = document.querySelector('.root')!;
+  private initialPage: MainPage;
+  private header: Header;
+  private footer: Footer;
+
+  static renderNewPage() {
+    const content = document.querySelector('.content')!;
+    if (content) {
+      content.innerHTML = '';
+    }
+    const hash = window.location.hash.slice(1);
+
+    let page: Page | null = null;
+    const arrIdPage = hash.split('/');
+
+    function getFilterHash(arrIdPage: Array<string>) {
+      const lastItem = arrIdPage[arrIdPage.length - 1];
+      switch(lastItem) {
```

Comment:
Here you don't use the benefits of switch case.
You can simply write smth like:
`if (lastItem === error) page = new ErrorPage(hash, ErrorTypes.Error404);` - create some condition for error
`else page = new MainPage(hash);`